src/lib/database/post.ts

```diff
@@ -73,7 +73,16 @@ export const getConversation = async (
       }
       currentPost = T;
     }
-    return posts;
+    // many posts to a single target, so gotta fetch them all!
```

Comment:
I just want to return a single post and have the other stuff happen in the background.

Reply:
I just want to return a single post and have the other stuff happen in the background.

Reply:
not sure i follow? this is used only to generate the agents replies and its already returning an array? how its currently implemented it wont fetch the whole conversation in the case that i have multiple responses to a single post?

Reply:
i guess is this a valid use case? https://github.com/johnhenry/antisocial-network/pull/4/files#diff-523f9b2351237923c551363e817ebab21d658a4dd7fc451ee35d6adb12bc8cfeR22

Reply:
would the example in the test not be considered a conversation? like i start with a post #1, a bot responds to that post (post #2), and the i add something to post #1 and ask a bot about the whole thing (post #3)? but posts #2 and #3 both would have post #1 as the target. should anything i ask the bot that targets post #1 not include all the related posts?

Reply:
A. I think I'm looking at this from last up , and you are looking at it from the the first down:

If you start at #1, you move up to targets, so the conversation would be just [#1]

If you start at #3, then the conversation would be [#3, #1]

If you start at #2, then the conversation would be [#2, #1]

I would consider [#3, #1] and  [#2, #1] to be different conversations. 

Currently there's a "/post merge" command that works similarly to the way you described in that it looks at a post and then looks at all of it's child conversations and creates a merged response. (see lib/database/post.ts/aggregatePostReplies)

(Note, i'm reworking how this operates).